src/ngraph/runtime/backend.hpp

```diff
@@ -109,4 +115,11 @@ class ngraph::runtime::Backend
     virtual bool is_supported_property(const Property prop) const;
 
     virtual void remove_compiled_function(std::shared_ptr<Executable> exec);
+
+    virtual std::shared_ptr<ngraph::runtime::Allocator> get_framework_memory_allocator();
```

Comment:
I don't understand what is a framework allocator and what is a device allocator. What is the difference? Is device memory something like pinned memory? Why does the framework need something special to allocate non-device memory beyond simple malloc?

Reply:
@rkimballn1 I think this is for the option if the framework wants to pass its to memory allocator to ngraph, so it can own the memory allocation and track the usage. @jbobba do we have any other specific use case where framework wants its allocator to be used in ngraph?

Reply:
In addition to tracking, frameworks might have better/high-perf allocators: e.g., https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/common_runtime/bfc_allocator.cc

Reply:
We could rename this to get_host_allocator if that is more informative.